CONTRIBUTING.md

```diff
@@ -135,56 +135,31 @@ In README files this ends up being slightly different. Because the version tag i
-  Changes are going to have to be made to `com.azure:azure-identity` first.
-  - [ ] Appropriately increase the version of `com.azure:azure-identity`
-  - [ ] Make the code changes
-  - [ ] Build and optionally install locally
-        This isn't completely necessary other than to install the updated version of the dependency into the local cache on the machine. The alternative to this would be to publish (DevOps or otherwise) and reference that version of the dependency after the release. Either one would allow `com.azure:azure-storage-common` to use the updated version of `com.azure:azure-identity`
-  - [ ] Submit the PR for the `com.azure:azure-identity`
-  - [ ] Merge the PR for the `com.azure:azure-identity`
-  - [ ] Publish the `com.azure:azure-identity` with the updated version.
-  
-  Only after the dependency `com.azure:azure-identity` has been published can the PR for `com.azure:azure-storage-common` be created.
-  - [ ] Appropriately increase the version of `com.azure:azure-storage-common` and the dependency version of `com.azure:azure-identity` in its pom file.
-  - [ ] Make the code changes, if any
-  - [ ] Build/Test or whatever
-  - [ ] Submit the PR for `com.azure:azure-storage-common`
-  - [ ] Merge the PR for `com.azure:azure-storage-common`
-  - [ ] Publish the PR for `com.azure:azure-storage-common`
+
+### Making changes to an already GA'd library that require other libraries to depend on the unreleased version
+
+This is where the `unreleased_` dependency tags come into play. Using the Unreleased Dependency example above, where `com.azure:azure-storage-blob` has a dependency on an unreleased `com.azure:azure-core`:
+
+- [ ] Make the additive changes to `com.azure:azure-core`
+- [ ] In [version_client.txt](./eng/versioning/version_client.txt) add the entry for the unreleased azure core in the unreleased section at the bottom of the file. The entry would look like `unreleased_com.azure:azure-core;<version>`.
+      Note: The version of the library referenced in the unreleased version tag should match the current version of that library.
+- [ ] In the pom.xml file for `com.azure:azure-storage-blob`, the dependency tag for `com.azure:azure-core` which was originally `{x-version-update;com.azure:azure-core-test;dependency}` would now become `{x-version-update;unreleased_com.azure:azure-core-test;dependency}`
+After the unreleased version of `com.azure:azure-core` was released but before `com.azure:azure-storage-blob` has been released.
+- [ ] In [version_client.txt](./eng/versioning/version_client.txt) the the dependency version of `com.azure:azure-core` would become the released version and the "unreleased_" entry, at this time, would be removed.
+- [ ] In the pom.xml file for `com.azure:azure-storage-blob`, the dependency tag for `com.azure:azure-core` would get changed back to `{x-version-update;com.azure:azure-core-test;dependency}`
 
```

Comment:
Could we automate this by hecking if `unreleased_` version is in available in maven -> break the build and it will force developer to remove "unreleased_" . May be create a separate ticket for this ?
May be I am asking too much. Ignore it if you feel it is unnecessory.

Reply:
I think the best we're going to get here is to verify that you're not trying to release something with "unreleaed_" dependencies. There's already a work item for that https://github.com/Azure/azure-sdk-for-java/issues/6184 (it's a two part work item, I've already got the first part completed).